src/devices/MAX31856/MAX31856.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Spi;
+using System.IO;
+using UnitsNet;
+
+namespace Iot.Device.MAX31856
+{
+    /// <summary>
+    /// Documentation https://datasheets.maximintegrated.com/en/ds/MAX31856.pdf
+    /// </summary>
+    public class MAX31856 : IDisposable
+    {
+        private readonly byte _thermocoupleType;
+        private SpiDevice _spiDevice;
+
+        #region SpiSettings
+
+        /// <summary>
+        /// Spi Clock Frequency from the Technical Manual of the device
+        /// </summary>
+        public const int SpiClockFrequency = 5_000_000;
+
+        /// <summary>
+        /// Spi Clock Frequency from the Technical Manual of the device
+        /// </summary>
+        public const int DataFlow = 0;
+
+        /// <summary>
+        /// MAX31856 SPI Mode
+        /// </summary>
+        public const SpiMode SpiMode = System.Device.Spi.SpiMode.Mode1;
+
+        #endregion
+
+        /// <summary>
+        /// Command to Get Temperature from the device
+        /// </summary>
+        /// <remarks>
+        /// Initializes the device and then reads the data next it converts the bytes to thermocouple temperature
+        /// </remarks>
+        public bool TryGetTemperature(out Temperature temperature)
+        {
+            try
+            {
+                temperature = ReadTemperature(out byte[] spiOutputData);
+                return true;
+            }
+            catch (IOException)
+            {
+                temperature = new Temperature();
+                return false;
+            }
+        }
+
+        /// <summary>
+        ///  Reads the temperature from the Cold-Junction sensor
+        /// </summary>
+        /// <returns>
+        /// Temperature, precision +- 0.7 Celsius range from -20 Celsius to +85 Celsius
+        /// </returns>
+        public Temperature GetColdJunctionTemperature() => Temperature.FromDegreesCelsius(ReadCJTemperature());
+
+        /// <summary>
+        /// Creates a new instance of the max31856.
+        /// </summary>
+        /// <param name="spiDevice">The communications channel to a device on a SPI bus</param>
+        /// <param name="thermocoupleType">Thermocouple Type T,K,R</param>
+        public MAX31856(SpiDevice spiDevice, ThermocoupleType thermocoupleType)
```

Comment:
Should you default thermocoupleType to something sensible so that user doesn't have to know about it? I personally have no clue what that means 😄

Reply:
Its typical with this device to use other thermocouple types such as K. Most common is T.